### PR TITLE
[Parser] Parse `local.get`

### PR DESCRIPTION
**Pull request by tlively**
This requires parsing local indices and fixing a bug in `Function::setLocalName`
where it only set up the mapping from index to name and not the mapping from
name to index.

### CODE REVIEW

**Comment by tlively** [comment]
Current dependencies on/for this PR:
* main
  * **PR #5137** <a href="https://app.graphite.dev/github/pr/WebAssembly/binaryen/5137" target="_blank">Graphite</a>  👈
    * **PR #5138** <a href="https://app.graphite.dev/github/pr/WebAssembly/binaryen/5138" target="_blank">Graphite</a> 
      * **PR #5139** <a href="https://app.graphite.dev/github/pr/WebAssembly/binaryen/5139" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/WebAssembly/binaryen/5137?utm_source=stack-comment).

**Review by kripken — APPROVED** [review]

**Comment by tlively** [comment]
**Graphite Merge Job**

Current status: 🔄 Waiting for checks to pass

This pull request is currently waiting for CI and mergeability checks to pass before being merged.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/WebAssembly/binaryen/5137?utm_source=stack-comment).


Job Reference: hKLpkfh0assiWMASwkXu

**Pull request merged**
